src/sentry/snuba/discover.py

```diff
@@ -195,8 +245,55 @@ def timeseries_query(selected_columns, query, params, rollup):
 
     This function is intended to only get timeseries based
     results and thus requires the `rollup` parameter.
+
+    Returns a SnubaTSResult object that has been zerofilled in
+    case of gaps.
+
+    selected_columns (Sequence[str]) List of public aliases to fetch.
+    query (str) Filter query string to create conditions from.
+    params (Dict[str, str]) Filtering parameters with start, end, project_id, environment,
+    rollup (int) The bucket width in seconds
+    reference (Reference) A reference event object. Used to generate additional conditions
+                          based on the provided reference.
+    referrer (str|None) A referrer string to help locate the origin of this query.
     """
-    raise NotImplementedError
+    snuba_filter = get_filter(query, params)
+
+    # Use fallbacks for date filters as zerofill() depends on dates.
+    snuba_args = {
+        "start": snuba_filter.start or datetime.utcfromtimestamp(0),
+        "end": snuba_filter.end or datetime.utcnow(),
```

Comment:
raw_query already fills the defaults with the retention period, you shouldn't need to do this here as well.

Reply:
But when do we ever want to fill all the way from `datetime.utcfromtimestamp(0)` anyway? This is just a placeholder that gets clamped to project retention later

Reply:
Good points. I'll add an exception for when the start/end are not defined. In practice that should never happen.